quipucords/api/common/entities.py

```diff
@@ -352,3 +380,27 @@ def hosts(self) -> List[HostEntity]:
     def number_of_hosts(self):
         """Retrieve the number of hosts."""
         return self.slice_end - self.slice_start
+
+
+class SystemPurposeRole(Enum):
+    """Representing system purpose role values accepted in HBI system profile."""
+
+    workstation = "Red Hat Enterprise Linux Workstation"
+    server = "Red Hat Enterprise Linux Server"
+    node = "Red Hat Enterprise Linux Compute Node"
+
+
+class SystemPurposeSla(Enum):
+    """Representing system purpose SLA values accepted in HBI system profile."""
+
+    premium = "Premium"
+    self_suport = "Self-Support"
+    standard = "Standard"
+
+
+class SystemPurposeUsage(Enum):
+    """Representing system purpose usage values accepted in HBI system profile."""
+
+    development_test = "Development/Test"
+    production = "Production"
+    disaster_recovery = "Disaster Recovery"
```

Comment:
mmm does inventory restrict the value for these fields?

Reply:
I believe so. Besides mentioning an enum list in the [inventory-schema](https://github.com/RedHatInsights/inventory-schemas/blob/cc4995e7b9e311919e55eb00578e2c4051c1727e/schemas/system_profile/v1.yaml#L526),  "Dev-Professional" appeared as a suitable value when calling `subscription-manager syspurpose service-level --list`. When I set my vm's system purpose sla to this value, insights report got to HBI but not to console. Observing kibana I could see an exception was raised during that specific field validation. This value was not on the enum list provided in the inventoy-schema.

Reply:
Cool! FYI we should the [spec at inventory repo](https://github.com/RedHatInsights/insights-host-inventory/blob/master/swagger/system_profile.spec.yaml) and [it's models.py module](https://github.com/RedHatInsights/insights-host-inventory/blob/master/app/models.py) as reference. 

In this case the values seem to be equal, so we are good.

Reply:
Only thing that bothers me is this code duplication across multiple repos. ugh.

Reply:
What do you mean by code duplication? I didnt quite get it what u mean.

Reply:
Not really, I meant the code I sent above. I tried it here and both commands have the same result. The vm is a RHEL8. 😶 
https://access.redhat.com/articles/system-purpose

Reply:
cool. I think I did something wrong on mine, then haha.